guides/expression_language.md

```diff
@@ -0,0 +1,171 @@
+# Trento Checks Expression Language Reference
+
+A small, fast, easy-to-use scripting language and evaluation engine
+
+- [Introduction](#introduction)
+- [Types](#corosync)
+- [Keywords](#keywords)
+- [Statements](#statements)
+- [Variables](#variables)
+- [Logic Operators](#logic-operators)
+- [If Statement](#logic-operators)
+- [Rhai]
+## Introduction
+
+[NameToBeDefined] is an embedded scripting language and evaluation engine for Trento Checks Expressions that gives a safe and easy way to script specific steps during Checks Execution.
+
+## Types
+TBD
+
+## Keywords
+TBD
+
+## Statements
+TBD
+
+## Variables
+TBD
+
+## Logic Operators
+
+| Operator | Description<br/>(`x` _operator_ `y`) | `x`, `y` same type or are numeric | `x`, `y` different types |
+| :------: | ------------------------------------ | :-------------------------------: | :----------------------: |
+|   `==`   | `x` is equals to `y`                 |       error if not defined        |  `false` if not defined  |
+|   `!=`   | `x` is not equals to `y`             |       error if not defined        |  `true` if not defined   |
+|   `>`    | `x` is greater than `y`              |       error if not defined        |  `false` if not defined  |
+|   `>=`   | `x` is greater than or equals to `y` |       error if not defined        |  `false` if not defined  |
+|   `<`    | `x` is less than `y`                 |       error if not defined        |  `false` if not defined  |
+|   `<=`   | `x` is less than or equals to `y`    |       error if not defined        |  `false` if not defined  |
+
+Comparison operators between most values of the same type are built in for all [standard types].
+
+Others are defined in the [`LogicPackage`][built-in packages] but excluded if using a [raw `Engine`].
+
+
+### Floating-point numbers interoperate with integers
+
+Comparing a floating-point number (`FLOAT`) with an integer is also supported.
+
+```rust
+42 == 42.0;         // true
+42.0 == 42;         // true
+42.0 > 42;          // false
+42 >= 42.0;         // true
+42.0 < 42;          // false
+```
+
+### Strings interoperate with characters
```

Comment:
I think this part is a bit too much complicated, and it also doesn't mention operations that are common when building a check such as arrays or maps operations.